xdsl/dialects/irdl/pyrdl_to_irdl.py

```diff
@@ -0,0 +1,91 @@
+from xdsl.builder import Builder
+from xdsl.dialects.irdl.irdl import (
+    AttributeOp,
+    DialectOp,
+    OperandsOp,
+    OperationOp,
+    ParametersOp,
+    ResultsOp,
+)
+from xdsl.ir.core import Block, Dialect, ParametrizedAttribute, Region, SSAValue
+from xdsl.irdl import (
+    AttrConstraint,
+    IRDLOperation,
+    OpDef,
+    ParamAttrDef,
+)
+
+from xdsl.dialects.irdl import AnyOp
+
+
+def constraint_to_irdl(builder: Builder, constraint: AttrConstraint) -> SSAValue:
+    """
+    Convert an attribute constraint to IRDL.
+    This will create new operations at the provided builder location.
+    """
+    any_op = builder.insert(AnyOp())
+    return any_op.output
+
+
+def op_def_to_irdl(op_def: type[IRDLOperation] | OpDef) -> OperationOp:
```

Comment:
feels like the caller could just call `cls.irdl_definition`, it's not much overhead